chore: Character

VulpoTheDev opened the pull request.

VulpoTheDev committed:
chore: character profile page

VulpoTheDev committed:
chore: added modals

VulpoTheDev committed:
chore: search bar and create folder modal

VulpoTheDev committed:
chore: Integrated Folders with API

kuroji-fusky committed:
chore: use `AsyncProps`

kuroji-fusky committed:
chore: updated logo

- updated footer links with semantic HTML

kuroji-fusky commented:
Logo has been officially updated 👍

VulpoTheDev commented:
Thank you!

VulpoTheDev committed:
chore: WIP Character Masthead

VulpoTheDev committed:
chore: add some spacing to tabs

VulpoTheDev committed:
feat: added html editor modal

VulpoTheDev committed:
feat: customizable panels

VulpoTheDev committed:
chore: added images

VulpoTheDev committed:
chore: display panels

VulpoTheDev committed:
feat: comments

VulpoTheDev committed:
chore: use asyncprops on params

kuroji-fusky committed:
chore: prettier format

- set `trailingComma` to `none`
- remove redudant imports

kuroji-fusky committed:
chore: refactored functions and types

- use the `cn` utility for `ThemeToggle
- refactored `displayer.ts`
- fixed deprecated type from `React.MutableRefObject`
- removed redudant `aria-disabled` attribute on `Checkbox.tsx` - the `disabled` attribute already includes this

kuroji-fusky commented on apps/web/src/utils/constants.ts:
```diff
@@ -1 +1,83 @@
+  { label: "Squirrel", value: "squirrel" },
+  { label: "Kangaroo", value: "kangaroo" },
+  { label: "Koala", value: "koala" },
+  { label: "Bat", value: "bat" },
+  { label: "Giraffe", value: "giraffe" },
+  { label: "Horse", value: "horse" },
+  { label: "Zebra", value: "zebra" },
+  { label: "Elephant", value: "elephant" },
+  { label: "Hippopotamus", value: "hippopotamus" },
+  { label: "Rhino", value: "rhino" },
+  { label: "Griffin", value: "griffin" },
+  { label: "Pegasus", value: "pegasus" },
+  { label: "Rat", value: "rat" },
+  { label: "Unicorn", value: "unicorn" }
+]
+
+export const pronounOptions = [
+  { label: "He/Him", value: "hehim" },
+  { label: "She/Her", value: "sheher" },
+  { label: "They/Them", value: "theythem" },
+  { label: "Ze/Hir", value: "zehir" },
+  { label: "Ze/Zir", value: "zezir" },
+  { label: "Xe/Xem", value: "xexem" },
+  { label: "Ey/Em", value: "eyem" },
+  { label: "Any Pronouns", value: "any" },
+  { label: "Prefer Not to Say", value: "prefernot" }
+]
+
+export const genderOptions = [
+  { label: "Male", value: "male" },
+  { label: "Female", value: "female" },
+  { label: "Non-Binary", value: "nonbinary" },
+  { label: "Genderfluid", value: "genderfluid" },
+  { label: "Agender", value: "agender" },
+  { label: "Transgender", value: "transgender" },
+  { label: "Genderqueer", value: "genderqueer" },
+  { label: "Two-Spirit", value: "twospirit" },
+  { label: "Prefer Not to Say", value: "prefernot" },
+  { label: "Other", value: "other" }
+]
```
While this is okay for an MVP, we'll most likely remove this in the future once we hit production since we can just fetch these data from the DB; then we add and remove at will; plus this is gonna significantly inflate the bundle size too, just a minor detail worth pointing out

kuroji-fusky replied:
Plus this is gonna significantly and the bundle too, just a minor detail worth pointing out

VulpoTheDev replied:
I gotchu, I went ahead and put a note on #60 so I can implement it in the backend

VulpoTheDev committed:
chore: dynamic panel loading

kuroji-fusky committed:
chore(deps): update `next` version

just a quick dep update due to a critical vulnerability - we should be fine since vercel takes care of everything

https://zhero-web-sec.github.io/research-and-things/nextjs-and-the-corrupt-middleware

VulpoTheDev committed:
feat: post comments, info panel

kuroji-fusky committed:
fix(minor): fixed typo on "nationality"

kuroji-fusky committed:
chore: distinguish type imports

kuroji-fusky committed:
minor: remove unused file

`mergeNamespaceComponents` is dormant and won't be used again

kuroji-fusky committed:
chore: housekeeping and formatting

kuroji-fusky committed:
chore: migrate ESLint and Prettier configs to Biome

kuroji-fusky committed:
minor: changed lint commands to biome

-  tweaking some biome settings

I didn't run any commands to prevent a merge conflict

kuroji-fusky committed:
refactor(actions): modified commands to biome

kuroji-fusky committed:
chore(deps): removed eslint and prettier

VulpoTheDev committed:
chore: replies on comments

VulpoTheDev committed:
Merge branch 'character' of https://github.com/MyArtverse-Project/MyArtverse into character

kuroji-fusky committed:
fix(imports): remove unused `mergeNamespaceComponents` import

VulpoTheDev committed:
chore: view replies function

VulpoTheDev committed:
chore: relationship modals

VulpoTheDev committed:
chore: ability to view profile logged out

kuroji-fusky committed:
Migrate from ESLint and Prettier to Biome (#67)

Removes both ESLint and Prettier's config files and dependencies; migrates all the configs from Biome

VulpoTheDev committed:
chore: improved profile visibility

VulpoTheDev committed:
Merge branch 'character' of https://github.com/MyArtverse-Project/MyArtverse into character

VulpoTheDev committed:
format: biome

VulpoTheDev committed:
chore: return null on silent errors

VulpoTheDev committed:
chore: display notifications

VulpoTheDev committed:
feat: search button

VulpoTheDev committed:
chore: wip search

VulpoTheDev committed:
chore: clickable links
